### PR TITLE
Artery G1: TCP framing foundation + envelope codec (gates on framing/envelope round-trip)

### PR DESCRIPTION
**Pull request by Aaronontheweb**
Implements milestone **G1** of [`artery-tcp-remoting`](https://github.com/akkadotnet/akka.net/tree/dev/openspec/changes/artery-tcp-remoting) (task groups 2–3): the TCP framing foundation and the Artery envelope codec, round-tripped in-memory and multi-segment. No async/lifecycle/transport yet (that's G2). Everything is `internal` — the public API surface is unchanged (`ApproveRemote` byte-identical).

## Design closures (first commit, design.md)

The envelope's open ◇ sub-decisions are closed, and the fixed header is revised **28 → 32 bytes** with an explicit **payload-offset** field: the draft derived the payload as `frame_length − header_length`, which breaks whenever a literal tail is present. The explicit offset makes the payload slice O(1) and position-independent. Closed: version = 1 (reject others); flags bit 0 = metadata section, bits 1–7 reserved-must-be-zero (reject); ABSENT tag = `0x0`; literals = u16 LE length + UTF-8 at absolute offsets ≥ 32 (< `0x00FFFFFF`, which is why the framing hard-caps frames at 16 MiB−1); metadata container = u32 LE length + bytes, skipped by the MVP decoder.

## Framing (`Akka.Remote.Artery`, internal)

- `ArteryStreamId` (control=1 / ordinary=2 / large=3), `ArteryConnectionHeader` (`AKKA` magic + stream id; tri-state parse: need-more-data / success / typed exception on bad magic or unknown id).
- `ArteryFrameParser`: incremental `[u32 LE length][body]` parser. Zero-copy — yields frame bodies as (possibly multi-segment) `ReadOnlySequence<byte>` over the appended chunks via a `ReadOnlySequenceSegment` chain, with a single-segment fast path so the common whole-frame-per-read case allocates no segment objects (this sits on the future serial decode island, Decision 2). Oversized frames fail fast from the declared length, before the body arrives.

## Envelope codec

- 32-byte fixed-offset header (version, flags, table versions, origin UID i64 — the 64-bit uid from #8317 — serializer id i32, three u32 tags, payload offset); allocation-free header decode with single-segment fast path; lazy literal/payload resolution; validation for version, reserved flags, tag/metadata/payload bounds.
- **V2 single-pass encode**: `FindSerializerV2For` + `ManifestFor` resolve serializer id + manifest *upfront* (`SerializerV2.Manifest(object)` is an upfront member; `SerializerV1Adapter` is itself a `SerializerV2`, so the path is uniform with no compat fork), header + literals are written first, the payload streams directly into a pooled growable `IBufferWriter<byte>` (`PooledFrameWriter`), and only the frame length + header are back-patched from actual bytes written. `SizeHint` is never trusted for framing.

## Verification

- 68 new unit tests (framing: fragmentation down to byte-by-byte delivery, multi-chunk frames, oversize/cap/boundary cases; envelope: ABSENT/LITERAL/mixed/UTF-8 round-trips, multi-segment decode with headers and literals split across segments, all rejection paths, compressed-tag surfacing, and an end-to-end V2 test: object → single-pass encode through real `Serialization` → decode → `Deserialize(ReadOnlySequence...)` → equality).
- Full Akka.Remote.Tests suite: 434 passed / 0 failed (net10.0, Release). `dotnet build -warnaserror` clean. API approvals unchanged.

**G1 gate: PASSED.** Next: G2 (handshake, association state, basic ordinary messaging over `Akka.Streams.IO.Tcp`).

### CODE REVIEW

File: src/core/Akka.Remote/Artery/ArteryConnectionHeader.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    /// Result of <see cref="ArteryConnectionHeader.TryParse"/>. Together with the exception
+    /// thrown for malformed input, this makes the parse outcome a tri-state:
+    /// <see cref="NeedMoreData"/> (not an error — the caller should append more bytes and
+    /// retry), <see cref="Success"/>, or an <see cref="ArteryFramingException"/> thrown
+    /// synchronously for a structurally invalid preamble (bad magic / unknown stream id).
+    /// </summary>
+    internal enum ArteryConnectionHeaderParseResult
+    {
+        /// <summary>Fewer than <see cref="ArteryConnectionHeader.Length"/> bytes were available; not an error.</summary>
+        NeedMoreData = 0,
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Artery/ArteryConnectionHeader.cs:27]: partial envelope result where we don't have enough data to fully parse the header

File: src/core/Akka.Remote/Artery/ArteryConnectionHeader.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        /// <paramref name="destination"/>.
+        /// </summary>
+        /// <param name="destination">
+        /// The destination span. Must be at least <see cref="Length"/> bytes long.
+        /// </param>
+        /// <param name="streamId">The stream identifier to write.</param>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="destination"/> is shorter than <see cref="Length"/>.
+        /// </exception>
+        public static void WriteTo(Span<byte> destination, ArteryStreamId streamId)
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Artery/ArteryConnectionHeader.cs:74]: LGTM

File: src/core/Akka.Remote/Artery/ArteryConnectionHeader.cs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            Span<byte> header = stackalloc byte[Length];
+            buffer.Slice(0, Length).CopyTo(header);
+
+            if (header[0] != MagicA || header[1] != MagicK || header[2] != MagicK || header[3] != MagicA)
+                throw new ArteryFramingException(
+                    $"Invalid Artery connection preamble: expected 'AKKA' magic bytes, got " +
+                    $"[0x{header[0]:X2} 0x{header[1]:X2} 0x{header[2]:X2} 0x{header[3]:X2}].");
+
+            var streamIdByte = header[StreamIdOffset];
+            streamId = streamIdByte switch
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Artery/ArteryConnectionHeader.cs:135]: worth noting: this identifies THE STREAM ID TYPE, not the unique instance of each stream. There can be multiple parallel lanes for ordinary streams.

File: src/core/Akka.Remote/Artery/ArteryEnvelopeCodec.cs [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+    internal static class ArteryEnvelopeCodec
+    {
+        /// <summary>
+        /// Total encoded size for an explicit-parts encode with the given sender/recipient/manifest
+        /// strings and payload length. Callers use this to rent a big-enough destination buffer
+        /// before calling <see cref="Encode(Span{byte},long,int,string?,string?,string,ReadOnlySpan{byte})"/>.
+        /// This is an exact size (not merely an upper bound) because, unlike the V2 single-pass
+        /// overload, every input here is already a concrete string/length.
+        /// </summary>
+        public static int MaxEncodedSize(string? senderPath, string? recipientPath, string manifest, int payloadLength)
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Artery/ArteryEnvelopeCodec.cs:43]: `serializerId` is a fixed-size payload that is always included, even though it's not represented here. This is mostly for the variable length elements.

**Review by Aaronontheweb — COMMENTED** [review]
LGTM
